resolve.go

```diff
@@ -107,3 +108,67 @@ func LookupServer(serverName ServerName) (*DNSResult, error) { // nolint: gocycl
 
 	return &result, nil
 }
+
+// ResolveServer implements the server name resolution algorithm described at
+// https://matrix.org/docs/spec/server_server/r0.1.1.html#resolving-server-names
+// Returns a slice containing the hosts (using the host:port form) that can be
+// used to send a federation request to the server using a given server name.
+// needWellKnown indicates whether a .well-known file should be looked up.
+// Returns an error if the server name isn't valid, or if either the .well-known
```

Comment:
failure to parse the wellKnown should not be treated as an error

Reply:
Oh that's true